### ISSUE TITLE
creating a grid of plots that are of different type?

### ISSUE THREAD

**Issue by an5ir** [pyviz/holoviews#3822]
is it possible to create a grid of plot of different types like below image:
![image](https://user-images.githubusercontent.com/42062013/61371142-ff615480-a88b-11e9-8a5e-3b749a16f84b.png)

i understand when using gridspace element have to be of the same type. so that does not work.
if i create each plot and put it into a layout, is there any way to leave the axis label on only if you are a plot at the edge of  a layout. without me having to manually go trough each plot an turning them on trough their option?

any help is much appreciated

**Comment by philippjfr** [comment]
@an5ir Have a look at the GridMatrix container. I've repurposed the issue to add docs about it to the reference gallery.

**Comment by an5ir** [comment]
Hi, are you referring to this example http://holoviews.org/gallery/demos/bokeh/iris_density_grid.html.
is the a way to individually choose which type of  plot you want for each square of the gridmatrix?

**Comment by philippjfr** [comment]
No, I'm referring to the hv.GridMatrix container object. The gridmatrix operation returns one of those, but the container is completely general purpose and you can put anything you like in it.

**Comment by an5ir** [comment]
thank you very much for your help. 

in terms  of the period - entropy square of the grid shown in blue,
![image](https://user-images.githubusercontent.com/42062013/61376931-18710200-a89a-11e9-9bac-4f18aea9f6ab.png)

is there any way to enter the 3 side by side horizontal bar plot in the grid matrix?
i understand it is not possible  making a layout of the 3 plot and then enter it into the gridmatrix, since does not accept layout.
do you know of a way i can combine the  3 graph into a single element and then enter it into the grid?

thanks again for the help.

**Comment by an5ir** [comment]
Hi, do you know if there is any way to carry this out?

Thanks

**Comment by poplarShift** [comment]
I think for complicated layouts such as these you're actually better of just building it as a standard Layout, sizing and modifiying components as you go...

If you want to do it with a GridMatrix, here is a start, but I'm not getting the sizes to align nicely.
```
import holoviews as hv
import numpy as np
hv.extension('bokeh')

np.random.seed(13)
x = np.random.rand(10,2)
el = hv.Scatter(x)
d = {
    ('a', 1): el,
    ('b', 1): el,
    ('a', 2): el,
    ('b', 2): el,
}
d2 = {
    ('c'): hv.GridMatrix({
        '3': el,
        '4': el,
    }, kdims='dim2').relabel(group='nested').opts(opts.Scatter(), opts.Layout(width=200))
}

l = hv.GridMatrix(d)

l[('b', 2)] = hv.GridMatrix(d2, kdims='dummy_dim')

l.opts(
    opts.Scatter(width=100, height=200), 
    opts.GridMatrix(
        toolbar=None, sizing_mode='scale_width',
    ),
)
```
